### DISCUSSION TITLE
Access other fields from within afterStateUpdated() method

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#1448] [Help] [answered]
When using the `afterStateUpdated()` method on a form field, is there a way to access the current value of another field? I'd like to be able to make changes to the `amount` field only if the `type` field has a certain value.

```
Forms\Components\TextInput::make('amount')
    ->numeric()
    ->afterStateUpdated(function (\Closure $set, $state) {
        $set('amount', $state);
    }),

```

**Comment by intrepidws** [comment]
To be more specific, this is on a create form. On an update form, I know that something like this would work, using the `$get` variable:

```
Forms\Components\TextInput::make('amount')
    ->numeric()
    ->afterStateUpdated(function (\Closure $set, \Closure $get, $state) {
        if ($get('type') == 'Refund') {
            $set('amount', -$state);
        } else {
            $set('amount', $state);
        }
    }),

```

However, in this instance `$get('type'` returns null (I assume because this is a create form and it isn't set yet).

**Comment by ryangjchandler** [comment]
@intrepidws Have you tried making both of the fields reactive?

**Comment by intrepidws** [comment, marked as answer]
Thanks @ryangjchandler, good idea. If I make the Amount field reactive it works great. Additionally, it seems that I can accomplish my desired result, without using `reactive` at all, if I swap `afterStateUpdated` with `dehydrateStateUsing`, which fires later in the lifecycle.
